### PR TITLE
8321941: Migrate test/jdk/valhalla/valuetypes from primitive classes to null-restricted types

### PR DESCRIPTION
**Pull request by mlchung**
Migrate test/jdk/valhalla/valuetypes tests to use null-restricted types instead.

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8321941](https://bugs.openjdk.org/browse/JDK-8321941): Migrate test/jdk/valhalla/valuetypes from primitive classes to null-restricted types (**Enhancement** - P3)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - Committer) ⚠️ Review applies to [d8683abd](https://git.openjdk.org/valhalla/pull/963/files/d8683abd186552f640d7845840619f23ee094e3b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla.git pull/963/head:pull/963` \
`$ git checkout pull/963`

Update a local copy of the PR: \
`$ git checkout pull/963` \
`$ git pull https://git.openjdk.org/valhalla.git pull/963/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 963`

View PR using the GUI difftool: \
`$ git pr show -t 963`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/963.diff">https://git.openjdk.org/valhalla/pull/963.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/valhalla/pull/963#issuecomment-1853127796)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java [hunk 1]
```diff
@@ -817,15 +821,22 @@ private static LambdaForm makePreparedFieldLambdaForm(byte formOp, boolean isVol
         final int U_HOLDER  = nameCursor++;  // UNSAFE holder
         final int INIT_BAR  = (needsInit ? nameCursor++ : -1);
         final int VALUE_TYPE = (hasValueTypeArg ? nameCursor++ : -1);
+        final int NULL_CHECK  = (isNullRestricted && !isGetter ? nameCursor++ : -1);
         final int PRE_CAST  = (needsCast && !isGetter ? nameCursor++ : -1);
         final int LINKER_CALL = nameCursor++;
+        final int FIELD_TYPE = (needsZeroInstance && isGetter ? nameCursor++ : -1);
+        final int ZERO_INSTANCE = (needsZeroInstance && isGetter ? nameCursor++ : -1);
         final int POST_CAST = (needsCast && isGetter ? nameCursor++ : -1);
         final int RESULT    = nameCursor-1;  // either the call or the cast
```
Comment (liach) [inline src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java:830]: Outdated comment, should be
```
// the call, the zero instance, or the cast
```

File: test/jdk/valhalla/valuetypes/ObjectMethods.java [hunk 2]
```diff
@@ -24,85 +24,128 @@
...
+    static final Point P2 = new Point(30, 40);
+    static final Line L1 = new Line(1, 2, 3, 4);
+    static final Line L2 = new Line(10, 20, 3, 4);
+    static final Ref R1 = new Ref(P1, L1);
+    static final Ref R2 = new Ref(P2, null);
+    static final Value V = new Value(P1, L1, R1, "value");
+
+    static Stream<Arguments> identitiesData() {
+        return Stream.of(
+                Arguments.of(new Object(), false, false),
```
Comment (RogerRiggs) [inline test/jdk/valhalla/valuetypes/ObjectMethods.java:119]: The JEP doesn't say this (yet) but `new Object()` is an identity instance. java.util.Objects.isIdentityObject() and .isValueObject() will need to be updated too.

File: test/jdk/valhalla/valuetypes/ArrayElementVarHandleTest.java [hunk 3]
```diff
@@ -24,116 +24,93 @@
 
 /*
  * @test
- * @summary test VarHandle on primitive class array
+ * @summary test VarHandle on value class array
  * @compile -XDenablePrimitiveClasses ArrayElementVarHandleTest.java
```
Comment (RogerRiggs) [inline test/jdk/valhalla/valuetypes/ArrayElementVarHandleTest.java:28]: It looks like the compiler does not need -XDenablePrimitiveClasses.
Except in RecursiveValueClass.java that still has a primitive class declaration.

Comment (mlchung) [reply test/jdk/valhalla/valuetypes/ArrayElementVarHandleTest.java]: Updated.

File: test/jdk/valhalla/valuetypes/ObjectMethods.java (hunk 2 above)
Comment (mlchung) [reply test/jdk/valhalla/valuetypes/ObjectMethods.java:118]: It already handles that.  The two arguments specify if the class of the instance is identity class or value class.     The test special cases to verify if the class is `Object.class`, it's an identity object.

File: test/jdk/valhalla/valuetypes/LambdaMetaFactory/LambdaTest.java [hunk 4]
```diff
@@ -23,62 +23,64 @@
 
 /*
  * @test
+ * @run junit/othervm -XX:+EnableValhalla LambdaTest
```
Comment (RogerRiggs) [inline test/jdk/valhalla/valuetypes/LambdaMetaFactory/LambdaTest.java:26]: Just style suggestion in the order of jtreg tags, \@summary should come before \@run

**Review by RogerRiggs — APPROVED** [review]

File: test/jdk/valhalla/valuetypes/LambdaMetaFactory/LambdaTest.java (hunk 4 above)
Comment (mlchung) [reply test/jdk/valhalla/valuetypes/LambdaMetaFactory/LambdaTest.java:26]: Hmm... we have a mixture of style.  The jtreg document states `@summary` comes before the action.   I will update them.

**Pull request closed without merging**